src/modules/__tests__/conditions.spec.js

```diff
@@ -0,0 +1,170 @@
+import { getConditions } from '../conditions.js'
+
+test('Legend set chosen with no conditions selected', () => {
+    const conditions = {
+        condition: '',
+        legendSet: 'legendSetId1',
+    }
+    const metadata = {
+        legendSetId1: { id: 'legendSetId1', name: 'Legend Set Name' },
+    }
+    const dimension = {}
+    const actual = getConditions({ conditions, metadata, dimension })
+
+    expect(actual).toEqual(['Legend Set Name'])
+})
+
+test('Legend set chosen with conditions selected', () => {
```

Comment:
```suggestion
test('Legend set chosen with legends selected', () => {
```

Reply:
Also change `{ id: 'Condition1Id', name: 'Condition 1' }` to ``{ id: 'Legend1Id', name: 'Legend 1' }`, as to not confuse conditions with legends.